src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -446,29 +410,26 @@ private void notifyLoginStatusChange(boolean login) {
     }
   }
 
-  private String queryEmail() {
-    try {
-      HttpRequest get = createRequestFactory().buildGetRequest(new GenericUrl(GET_EMAIL_URL));
-      HttpResponse resp = get.execute();
-
-      String responseString = "";
-      try (Scanner scan = new Scanner(resp.getContent())) {
-        while (scan.hasNext()) {
-          responseString += scan.nextLine();
-        }
-      }
-
-      String userEmail = parseUrlParameters(responseString).get("email");
-      if (userEmail == null) {
-        loggerFacade.logWarning("Could not parse email after Google service sign-in");
+  @VisibleForTesting
+  String queryEmail(Credential credential) throws IOException {
+    HttpRequest get = createRequestFactoryInternal(credential)
+        .buildGetRequest(new GenericUrl(emailQueryUrl))
+        .setConnectTimeout(5000 /* ms */)
+        .setReadTimeout(5000 /* ms */);
+    HttpResponse resp = get.execute();
+
+    String responseString = "";
+    try (Scanner scan = new Scanner(resp.getContent())) {
+      while (scan.hasNext()) {
+        responseString += scan.nextLine();
       }
-      return userEmail;
+    }
 
-    } catch (IOException ioe) {
-      // catch exception in case something goes wrong in parsing the response
-      loggerFacade.logError("Could not parse email after Google service sign-in", ioe);
-      return null;
+    String userEmail = parseUrlParameters(responseString).get("email");
+    if (userEmail == null) {
+      throw new IOException("Could not parse email after Google service sign-in");
```

Comment:
I felt that way too. Let me throw `EmailAddressNotReturnedException` instead of `IOException`.